lib/executor/executor.cpp

```diff
@@ -206,5 +207,190 @@ Executor::asyncInvoke(const Runtime::Instance::FunctionInstance *FuncInst,
+  spdlog::info("Collecting memory section");
+
+  AST::MemorySection MemSec;
+  const auto *CurrentInstance = StackMgr.getModule();
+  auto MemInstances = CurrentInstance->getMemoryInstances();
+  std::cout << MemInstances.size() << "\n";
+  std::cout << MemInstances[0]->getMemoryType().getLimit().hasMax() << "\n";
+  std::cout << MemInstances[0]->getMemoryType().getLimit().getMax() << "\n";
+  std::cout << MemInstances[0]->getMemoryType().getLimit().getMin() << "\n";
+  MemInstances[0]->getMemoryType().getLimit();
+  // In the current version of WebAssembly, at most one memory may be defined
+  // or imported in a single module, and all constructs implicitly reference
+  // this memory.
+  //
+  // If hasMax, set Max to that.
+  // Else set whatever Min exists
+
+  return MemSec;
+}
+
+// Data section
+// TODO malformed section
+AST::DataSection Executor::collectDataSection(Runtime::StackManager &StackMgr) {
+  spdlog::info("Collecting Memories");
+
+  AST::DataSection Datasec;
+
+  const auto *CurrentInstance = StackMgr.getModule();
+  auto &DataInstances = CurrentInstance->getOwnedDataInstances();
+
+  AST::DataSegment Seg;
+  auto &Content = Seg.getData();
+
+  for (auto &Iter : DataInstances) {
+    Content.insert(Content.end(), Iter->getData().begin(),
+                   Iter->getData().end());
+  }
+  if (Content.size() == 0) {
+    std::cout << "empty\n";
+    std::cout << Content.max_size();
+    Content.insert(Seg.getData().end(), 0x00);
+  }
+  std::cout << Content.size();
+  Datasec.getContent().push_back(Seg);
+  return Datasec;
+}
+
+// corestack   ::= customsec(thread-info vec(frame))
+// thread-info ::= 0x0 thread-name:name
+// frame       ::= 0x0 funcidx:u32 codeoffset:u32
+//                     locals:vec(value) stack:vec(value)
+AST::CustomSection Executor::collectCoreStack(Runtime::StackManager &StackMgr) {
+  spdlog::info("Constructing CoreStack");
+
+  AST::CustomSection CoreStack;
+  CoreStack.setName("corestack");
+
+  auto Frames = StackMgr.getAllFrames();
+  std::reverse(Frames.begin(), Frames.end());
+  auto Size = Frames.size();
+
+  // TODO get thread info
+  auto &Content = CoreStack.getContent();
+  Content.push_back(0x00);
+  Content.push_back(0x04);
+
+  std::string str = "main";
+  Content.insert(Content.end(), str.begin(), str.end());
+  for (size_t i = 0; i < Size; i++) {
+    if (Frames[i].Module == nullptr) // guard
+      continue;
+
+    auto Funcidx = Frames[i].From->getTargetIndex(); // function index
+    // TODO which to use?
+    //  auto Funcidx = Item.From->getTargetIndex(); // function index
+    auto Codeoffset = 0;
+    // auto Codeoffset = 0; // only if codeoffset is unknown?
+    auto LocalsN = Frames[i].Locals;
+    std::cout << Frames[i].From->getOffset() << "\n";
+
+    auto Locals = StackMgr.getRangeSpan(Frames[i].VPos, Frames[i].VPos);
+    auto Stack =
+        StackMgr.getTopSpan(StackMgr.size() - LocalsN); // TODO is this correct?
```

Comment:
All frames are using the same stack. `getTopSpan` will not give you the correct value stack. Please follow the previous review about the stack layout and do the proper calculation.